File: 1557/minimum-number-of-vertices-to-reach-all-nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        int [] inN = new int[n];
        int [] outN = new int[n];
        for(List<Integer> edge : edges){
            outN[edge.get(0)]++;
            inN[edge.get(1)]++;
        }
        List<Integer> ans = new ArrayList<Integer>();
        for(int i=0; i<n; i++){
            if(inN[i]==0) ans.add(i);
        }
        return ans;
    }
}

File: 1557/minimum-number-of-vertices-to-reach-all-nodes.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findSmallestSetOfVertices(int n, int** edges, int edgesSize, int* edgesColSize, int* returnSize){
    int in[n];
    for(int i=0; i<n; i++) in[i] = 0;

    for(int i=0; i<edgesSize; i++) {
        int from = edges[i][0];
        int to = edges[i][1];
        in[to]++;
    }

    int ansN=0;
    for(int i=0; i<n; i++){
        if(in[i]==0) ansN++;
    }

    int * ans = (int*) malloc(sizeof(int)*ansN);
    ansN = 0;
    for(int i=0; i<n; i++){
        if(in[i]==0) ans[ansN++] = i;
    }
    *returnSize = ansN;
    return ans;
}
